packages/core/src/indexing-store/profile.ts

```diff
@@ -0,0 +1,351 @@
+import type { Event } from "@/internal/types.js";
+import type { Column, Table } from "drizzle-orm";
+import type { ProfilePattern, Row } from "./cache.js";
+import { getCacheKey } from "./utils.js";
+
+export const getProfilePatternKey = (pattern: ProfilePattern): string => {
+  return Object.values(pattern).join("_");
+};
+
+const eq = (
+  target: bigint | string | number | boolean | null | undefined,
+  value: bigint | string | number | boolean | null | undefined,
+) => {
+  if (!target) return false;
```

Comment:
Worth double checking this boolean logic, or perhaps commenting which type combos each of these cases handle